week-5-practise/js/main.js

```diff
@@ -0,0 +1,76 @@
+
+/////Blue button Click ///////
+
+var blueBtn = document.querySelector("#blueBtn"); // getting blue button
+blueBtn.addEventListener("click",doBlueStuff);
+var blueBtnClass = document.querySelector(".jumbotron"); // gettimg jumbotron class
+var divBtn = document.querySelector(".buttons"); // getting div buttons class
+
+
+function doBlueStuff(){    
+    blueBtnClass.style.backgroundColor = "#588fbd";
+//////////////donate a bike button changes////////////////
+    
+    divBtn.firstElementChild.style.backgroundColor = "#ffa500";
+//////////////// change to  Volunteer Button//////////////////
+    divBtn.lastElementChild.style.backgroundColor = "black";
+    divBtn.lastElementChild.style.color = "white";
+}
+
+
+/////////////////////// click on orange button /////////////
+var orangeBtnClick = document.querySelector("#orangeBtn");
+orangeBtnClick.addEventListener("click",doOrangeStuff)
+
+function doOrangeStuff(){
+blueBtnClass.style.backgroundColor = "#f0ad4e";
+divBtn.firstElementChild.style.backgroundColor = "#5751fd";
+divBtn.lastElementChild.style.backgroundColor = "#31b0d5";
+divBtn.lastElementChild.style.color = "white";
+}
+///////////Click om Green Button //////////////
+var greenBtnClick = document.querySelector("#greenBtn");
+greenBtnClick.addEventListener("click",doGreenStuff);
+
+function doGreenStuff(){
+blueBtnClass.style.backgroundColor = "#87ca8a";
+divBtn.firstElementChild.style.backgroundColor = "black";
+divBtn.lastElementChild.style.backgroundColor = "#8c9c08";
+}
+
+
+/////// When Submit is Clicked ////////
+var formTag = document.querySelector("form");
+    formTag.lastElementChild.addEventListener("click",validateForm); //getting submit button
+    
+    function validateForm() {
+        var emailInput = document.querySelector("#exampleInputEmail1");
+        var nameInput = document.querySelector("#example-text-input");
+        var describeMe = document.querySelector("#exampleTextarea");
+        event.preventDefault();
+        if (emailInput.value.length === 0){
+            emailInput.style.backgroundColor = "red"
+        }
+        else if (nameInput.value.length === 0){
+            nameInput.style.backgroundColor = "red";
+        }
+        else if (describeMe.value.length === 0){
+            describeMe.style.backgroundColor = "red";
+
+        }
+         else if (!emailInput.value.includes("@")){
+            emailInput.style.backgroundColor = "red";
+         }
+            
```

Comment:
You've got the basic mechanisms working: checking values and setting red colour accordingly - well done.  However, there are a couple of bugs in this `validateForm` function, mostly with your logic.

Here are some example issues:

1) If I load the page and enter just "a" in email, and click submit, it highlights only the 'name' field in red, but the specification says:
`For all the fields that invalid, it should make their background color 'red'.`, so in this case, ALL fields should be highlit red, not just one!

2) If I then go on to put a valid name in the red name field and resubmit, the name field stays red, indicating there's still an error with it, but there's not.

Have a play with the form, and see for yourself that these issues exist.  Then try to fix the issues, so that the form behaves as specified and as you would expect.